Exercise_4.py

```diff
@@ -7,15 +7,37 @@ def __init__(self, data):
         self.right = None
           
 """ Inorder traversal of a binary tree"""
-def inorder(temp): 
-  
-    
+def inorder(temp):
+    if temp:
+        inorder(temp.left)
+        print(temp.key, end=" ")
+        inorder(temp.right)
+#runtime complexity O(n)
   
   
 """function to insert element in binary tree """
 def insert(temp,key): 
-  
-    
+    #Assuming NO DUPLICATES are allowed in binary key 
+    current = temp
+    while current.left:
```

Comment:
No, this is not the right way to insert in a Binary tree, this will end up forming a linked list like Data Structure where only the left pointers are populated. Please redo this exercise, You can reach out on Slack if you have doubts.